scripts/SANS/sans/algorithm_detail/batch_execution.py

```diff
@@ -1134,8 +1135,20 @@ def get_all_names_to_save(reduction_packages, save_can):
 
         transmission = reduction_package.unfitted_transmission
         transmission_can = reduction_package.unfitted_transmission_can
-        trans_name = '' if not transmission else transmission.name()
-        transCan_name = '' if not transmission_can else transmission_can.name()
+        try:
+            # Does not always work for event slice data as transmission
+            # workspaces can get deleted.
+            # This is a temporary fix to allow file saving with
+            # event sliced data.
+            trans_name = '' if not transmission else transmission.name()
+        except Exception:
```

Comment:
Might it be useful here to pass the exception type and message to the logger in debug? Since this is a temporary fix it should help the next developer with the permanent fix.